### PR TITLE
[Symfony] Add dump-env task

### PR DESCRIPTION
**Pull request by richardhj**
Add task to run dump-env on deployment

- [ ] Bug fix #…?
- [x] New feature?
- [ ] BC breaks?
- [ ] Tests added?
- [x] Docs added?

      Please, regenerate docs by running next command:
      $ php bin/docgen

As recommended on https://symfony.com/doc/current/deployment.html#b-configure-your-environment-variables

Run `composer dump-env` on every deployment.

The `$APP_ENV` shall be automatically when using `set('dotenv', '{{deploy_path}}/shared/.env.local');` in the deploy.php file.

### CODE REVIEW

**Comment by richardhj** [comment]
Kindly request review from @garak @Tobion

**Review by garak — APPROVED** [review]

File: recipe/symfony.php [hunk 1]
```diff
@@ -62,6 +62,13 @@
...
 
+desc('Optimize environment variables');
+task('deploy:env', function () {
+    within('{{release_or_current_path}}', function () {
+        run('{{bin/composer}} dump-env "${APP_ENV:-prod}"');
+    });
+});
+
 desc('Deploys project');
 task('deploy', [
```
Comment (Seb33300) [inline recipe/symfony.php:73]: If you want to dump env on every deployment, you should add it to the `deploy` task?

**Review by Seb33300 — APPROVED** [review]

**Comment by mrhackcz** [comment]
@richardhj Was it good idea? How can you do `dump-env` before `composer install` without symfony/flex ?

```
deploy:update_code
deploy:env
[development]  error  in symfony.php on line 69:
[development] run cd /www/hosting/xxxx/yyyy/releases/1 && (/usr/local/bin/composer dump-env "${APP_ENV:-prod}")
[development] err Command "dump-env" is not defined.
[development] exit code 1 (General error)
ERROR: Task deploy:env failed!
```

**Comment by richardhj** [comment]
> @richardhj Was it good idea? How can you do `dump-env` before `composer install` without symfony/flex ?
> 
dump-env is a symfony/flex utility - maybe that's the issue here?

**Comment by mrhackcz** [comment]
> > @richardhj Was it good idea? How can you do `dump-env` before `composer install` without symfony/flex ?
> 
> dump-env is a symfony/flex utility - maybe that's the issue here?

Yeah, but now, `deploy:env` in `symfony.php` is rewriting the default `deploy:env` in common.php. That now starts even before `deploy:vendors` which is installing symfony/flex.

**Comment by richardhj** [comment]
Ok, this is an ugly naming conflict that was unnoticed until now because the Symfony recipe in fact overrides the common.php's deploy task (in Symfony recipes, there was no deploy:env before).

**Comment by YetiCGN** [comment]
> > > @richardhj Was it good idea? How can you do `dump-env` before `composer install` without symfony/flex ?
> > 
> > 
> > dump-env is a symfony/flex utility - maybe that's the issue here?
> 
> Yeah, but now, `deploy:env` in `symfony.php` is rewriting the default `deploy:env` in common.php. That now starts even before `deploy:vendors` which is installing symfony/flex.

Which is actually a problem for us: https://github.com/deployphp/deployer/discussions/3938

**Pull request merged**